src/shared/seccomp-util.c

```diff
@@ -878,25 +876,26 @@ int seccomp_add_syscall_filter_item(scmp_filter_ctx *seccomp, const char *name,
                         return -EINVAL;
                 }
 
-                r = seccomp_add_syscall_filter_set(seccomp, other, action, exclude);
-                if (r < 0)
-                        return r;
+                return seccomp_add_syscall_filter_set(seccomp, other, action, exclude);
+
         } else {
-                int id;
+                int id, r;
 
                 id = seccomp_syscall_resolve_name(name);
                 if (id == __NR_SCMP_ERROR) {
                         log_debug("System call %s is not known, ignoring.", name);
                         return 0;
                 }
 
+                if (id < 0)
+                        /* No such system call exists on this architecture. */
+                        return 0;
+
                 r = seccomp_rule_add_exact(seccomp, action, id, 0);
                 if (r < 0)
-                        /* If the system call is not known on this architecture, then that's fine, let's ignore it */
-                        log_debug_errno(r, "Failed to add rule for system call %s() / %d, ignoring: %m", name, id);
+                        return log_debug_errno(r, "Failed to add rule for system call %s() / %d, ignoring: %m", name, id);
+                return 0;
```

Comment:
This seems to change the behavior.

Reply:
We were ignoring all errors, and that means that "legitimate" errors could be hidden in between the trivial errors about unimplemented syscalls. I'll wait for CI this time.

> BTW, why do you propagate the error here, but not in `seccomp_load_syscall_filter_set_raw`()? It seems not consistent for me...

Yep, I'll change `seccomp_load_syscall_filter_set_raw` too.